### PR TITLE
Update list-item-selector schema

### PR DESCRIPTION
**Pull request by berroar**
### PR Context
This PR adds support for the new feature in eq-questionnaire-runner to resolve answer values based on the first item in a given list, by expanding on the `list_item_selector` field.  [see runner PR](https://github.com/ONSdigital/eq-questionnaire-runner/pull/126)

### Checklist
Check that nothing is broken by this change

### CODE REVIEW

**Comment by berroar** [comment]
[[K] Feature to allow placeholders to resolve the first item in a list (M)](https://trello.com/c/neW1nxP5/3787-k-feature-to-allow-placeholders-to-resolve-the-first-item-in-a-list-m)

**Comment by iwootten** [comment]
We should be accompanying schema changes with unit tests to demonstrate the feature.

File: schemas/string_interpolation/definitions.json [hunk 1]
```diff
@@ -160,12 +160,11 @@
...
             "description": "The source of the data.",
             "type": "string",
-            "enum": ["location"]
+            "enum": ["location", "list"]
           },
           "id": {
             "description": "The unique id of the object.",
-            "type": "string",
-            "enum": ["list_item_id", "to_list_item_id"]
+            "type": "string"
```
Comment (ajmaddaford) [inline schemas/string_interpolation/definitions.json:167]: This should restrict the combinations allowed. So only `list_item_id` and `to_list_item_id` are  allowed for `location`, and `first` and `primary_person` for `list`.

File: tests/schemas/valid/test_placeholder_based_on_first_item_in_list.json [hunk 2]
```diff
@@ -0,0 +1,367 @@
+{
+  "mime_type": "application/json/ons/eq",
+  "language": "en",
+  "schema_version": "0.0.1",
+  "data_version": "0.0.3",
+  "survey_id": "0",
+  "title": "Repeating Section Summaries",
+  "theme": "default",
+  "description": "A questionnaire to repeating section summary functionality",
```
Comment (iwootten) [inline tests/schemas/valid/test_placeholder_based_on_first_item_in_list.json:9]: This description doesn't make sense

**Pull request merged**